src/components/book-form.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import * as actions from '../actions/index';
+
+const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+let identifier = 0;
+
+class BookForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+  }
+
+  handleFormSubmit(e) {
+    e.preventDefault();
+    const { handleAdding } = this.props;
+    const book = {
+      id: identifier,
+      title: this.titleInput.value,
+      category: categories[this.categoryValue.options.selectedIndex],
+    };
+    identifier += 1;
+    handleAdding(book);
+    this.titleInput.value = '';
+    this.categoryValue.options.selectedIndex = 0;
+  }
+
+  render() {
+    return (
+      <form>
+        <div>
+          <label htmlFor="title">Book Title</label>
+          <input ref={(node) => { this.titleInput = node; }} type="text" id="title" />
+        </div>
+        <div>
+          <label htmlFor="ctg">Book Title</label>
+          <select name="category" id="ctg" ref={(node) => { this.categoryValue = node; }}>
+            {categories.map((item) => <option key={item} value={item}>{item}</option>)}
+          </select>
+        </div>
+        <button type="submit" onClick={this.handleFormSubmit}> Submit </button>
```

Comment:
Its better to use a submit event on the form. **More on that in the next milestone**